inc/dropdown.class.php

```diff
@@ -3002,54 +3051,53 @@ public static function getDropdownFindNum($post, $json = true) {
          return;
       }
 
-      if ($item->isEntityAssign()) {
-         if (isset($post["entity_restrict"]) && ($post["entity_restrict"] >= 0)) {
-            $entity = $post["entity_restrict"];
-         } else {
-            $entity = '';
-         }
-
-         // allow opening ticket on recursive object (printer, software, ...)
-         $recursive = $item->maybeRecursive();
-         $where     = getEntitiesRestrictRequest("WHERE", $post['table'], '', $entity, $recursive);
-
-      } else {
-         $where = "WHERE 1";
-      }
-
+      $where = [];
       if (isset($post['used']) && !empty($post['used'])) {
-         $where .= " AND `id` NOT IN ('".implode("','", $post['used'])."') ";
+         $where['NOT'] = ['id' => $post['used']];
       }
 
       if ($item->maybeDeleted()) {
-         $where .= " AND `is_deleted` = 0 ";
+         $where['is_deleted'] = 0;
       }
 
       if ($item->maybeTemplate()) {
-         $where .= " AND `is_template` = 0 ";
+         $where['is_template'] = 0;
       }
 
       if (isset($_POST['searchText']) && (strlen($post['searchText']) > 0)) {
-         $search = Search::makeTextSearch($post['searchText']);
-
-         $where .= " AND (`name` ".$search."
-                        OR `id` = '".$post['searchText']."'";
+         $search = Search::makeTextSearchValue($post['searchText']);
+         $orwhere =[
+            'name'   => $search,
+            'id'     => $post['searchText']
+         ];
 
          if ($DB->fieldExists($post['table'], "contact")) {
-            $where .= " OR `contact` ".$search;
+            $orwhere['contact'] = $search;
          }
          if ($DB->fieldExists($post['table'], "serial")) {
-            $where .= " OR `serial` ".$search;
+            $orwhere['serial'] = $search;
          }
          if ($DB->fieldExists($post['table'], "otherserial")) {
-            $where .= " OR `otherserial` ".$search;
+            $orwhere['otherserial'] = $search;
          }
-         $where .= ")";
+         $where['AND'] = ['OR' => $orwhere];
```

Comment:
'AND' is useless here
```suggestion
         $where[] = ['OR' => $orwhere];
```